File: db/holiday.py
from db import Database

class HolidayDB(Database):
    def __init__(self):
        super().__init__()
        self.create()

    def create(self):
        self.execute('''CREATE TABLE IF NOT EXISTS holidays
                     (MONTH INT NOT NULL,
                     DAY INT NOT NULL,
                     MSG VARCHAR(2000) NOT NULL)''')
        
    def add(self, month, day, msg):
        data = self.execute("SELECT * FROM holidays WHERE MONTH={} AND DAY={}".format(month,day))
        updated = False
        if len(data) == 0:
            sql = "INSERT INTO holidays (MONTH, DAY, MSG) VALUES ({},{},\"{}\")".format(month,day,msg)
        else:
            updated = True
            sql = "UPDATE holidays SET MSG='{}' WHERE MONTH={} AND DAY={}".format(msg, month, day)
        self.execute(sql)
        return updated

    def check(self, month, day):
        data = self.execute("SELECT MSG FROM holidays WHERE MONTH={} AND DAY={}".format(month,day))
        if len(data) == 0:
            return 0
        else:
            return data[0][0]

    def check_multi(self):
        return self.execute("SELECT * FROM holidays")

    def remove(self, month, day):
        data = self.execute("SELECT * FROM holidays WHERE MONTH={} AND DAY={}".format(month,day))
        if len(data) == 0:
            return 0
        else:
            self.execute("DELETE FROM holidays WHERE MONTH={} AND DAY={}".format(month,day))
            return 1

File: db/holiday.js
const logger = require('../utility/logger');
const db = require('./database');

class HolidayDB{
  constructor() {
    this.create();
  }
  async create() {
    logger.info('Checking/creating holidays table');
    await db.execute(`CREATE TABLE IF NOT EXISTS holidays
                        (MONTH INT NOT NULL,
                        DAY INT NOT NULL,
                        MSG VARCHAR(2000) NOT NULL)`);
  }

  async add(month, day, msg) {
    let updated = false;
    let sql = '';
    let params = []
    const data = await db.execute('SELECT * FROM holidays WHERE MONTH=? AND DAY=?', [month, day]);
    if (data) {
      logger.info(`Adding holiday on ${month}/${day} with ${msg} to holidays table`);
      sql = `INSERT INTO holidays (MONTH, DAY, MSG) VALUES (?, ?, '?')`;
      params = [month, day, msg];
    }
    else {
      updated = true;
      logger.info(`Updated holiday on ${month}/${day} with ${msg} to holidays table`);
      sql = 'UPDATE holidays SET MSG=\'?\' WHERE MONTH=? AND DAY=?';
      params = [msg, month, day];
    }
    await db.execute(sql, params);
    return updated;
  }

  async check(month, day) {
    logger.info(`Checking for holiday on ${month}/${day}`);
    const data = await db.execute('SELECT MSG FROM holidays WHERE MONTH=? AND DAY=?', [month, day]);
    if (!db.checkLen(data)) {
      return 0;
    }
    else {
      return data[0].MSG;
    }
  }

  async checkMulti() {
    logger.info('Getting all holidays');
    return await db.execute('SELECT * FROM holidays');
  }

  async remove(month, day) {
    logger.info(`Removing holiday on ${month}/${day} from the holidays table`);
    const data = await db.execute('SELECT * FROM holidays WHERE MONTH=? AND DAY=?', [month, day]);
    if (data.length == 0) {
      return 0;
    }
    else {
      await db.execute('DELETE FROM holidays WHERE MONTH=? AND DAY=?', [month, day]);
      return 1;
    }
  }
}

module.exports = new HolidayDB();